File: model/video_subtitle_erase.go
package model

type VideoSubtitleEraseJobStatus string

const (
	VideoSubtitleEraseJobStatusQueued      VideoSubtitleEraseJobStatus = "queued"
	VideoSubtitleEraseJobStatusUploading   VideoSubtitleEraseJobStatus = "uploading"
	VideoSubtitleEraseJobStatusProcessing  VideoSubtitleEraseJobStatus = "processing"
	VideoSubtitleEraseJobStatusDownloading VideoSubtitleEraseJobStatus = "downloading"
	VideoSubtitleEraseJobStatusSucceeded   VideoSubtitleEraseJobStatus = "succeeded"
	VideoSubtitleEraseJobStatusFailed      VideoSubtitleEraseJobStatus = "failed"
)

type VideoSubtitleEraseJob struct {
	ID                       string                      `json:"id" gorm:"primaryKey"`
	UserID                   string                      `json:"-" gorm:"index"`
	ProjectID                string                      `json:"projectId" gorm:"index"`
	CanvasID                 string                      `json:"canvasId" gorm:"index"`
	SourceNodeID             string                      `json:"sourceNodeId"`
	SourceAssetID            string                      `json:"sourceAssetId"`
	Provider                 string                      `json:"provider"`
	InputTOSURL              string                      `json:"-" gorm:"type:text"`
	RunID                    string                      `json:"runId" gorm:"index"`
	ClientToken              string                      `json:"-" gorm:"index"`
	ProviderRequestID        string                      `json:"providerRequestId"`
	ProcessingStage          string                      `json:"processingStage"`
	Status                   VideoSubtitleEraseJobStatus `json:"status" gorm:"index"`
	Progress                 int                         `json:"progress"`
	Attempt                  int                         `json:"attempt"`
	InputWidth               int                         `json:"inputWidth"`
	InputHeight              int                         `json:"inputHeight"`
	InputDurationSeconds     float64                     `json:"inputDurationSeconds"`
	InputMIMEType            string                      `json:"inputMimeType"`
	InputBytes               int64                       `json:"inputBytes"`
	InputPath                string                      `json:"-" gorm:"type:text"`
	OutputWidth              int                         `json:"outputWidth"`
	OutputHeight             int                         `json:"outputHeight"`
	OutputDurationSeconds    float64                     `json:"outputDurationSeconds"`
	EstimatedBillableMinutes float64                     `json:"estimatedBillableMinutes"`
	EstimatedCostCNY         float64                     `json:"estimatedCostCny"`
	PricingRuleVersion       string                      `json:"pricingRuleVersion"`
	CostEstimateAvailable    bool                        `json:"costEstimateAvailable"`
	ResultSourceURL          string                      `json:"-" gorm:"type:text"`
	ResultURL                string                      `json:"resultUrl" gorm:"type:text"`
	ResultMIMEType           string                      `json:"resultMimeType"`
	ResultBytes              int64                       `json:"resultBytes"`
	ErrorCode                string                      `json:"errorCode"`
	ErrorMessage             string                      `json:"errorMessage" gorm:"type:text"`
	CloudProcessing          bool                        `json:"cloudProcessing"`
	CreatedAt                string                      `json:"createdAt" gorm:"index"`
	StartedAt                string                      `json:"startedAt"`
	CompletedAt              string                      `json:"completedAt"`
	UpdatedAt                string                      `json:"updatedAt"`
}

File: web/src/services/api/video-subtitle-erase.ts
import { useUserStore } from "@/stores/use-user-store";
import { apiGet, apiPostEmpty, apiPostForm } from "./request";

export type VideoSubtitleEraseJobStatus = "queued" | "uploading" | "processing" | "downloading" | "succeeded" | "failed";

export type VideoSubtitleEraseCapabilities = {
    enabled: boolean;
    provider: string;
    maxInputBytes: number;
    maxInputWidth: number;
    maxInputHeight: number;
    maxOutputWidth: number;
    maxOutputHeight: number;
    outputFormat: "mp4";
    cloudProcessing: true;
    pricing: { unitPriceCny: number; ruleVersion: string };
};

export type VideoSubtitleEraseJob = {
    id: string;
    projectId: string;
    canvasId: string;
    sourceNodeId: string;
    sourceAssetId: string;
    provider: string;
    runId: string;
    providerRequestId: string;
    processingStage: string;
    status: VideoSubtitleEraseJobStatus;
    progress: number;
    attempt: number;
    inputWidth: number;
    inputHeight: number;
    inputDurationSeconds: number;
    inputMimeType: string;
    inputBytes: number;
    outputWidth: number;
    outputHeight: number;
    outputDurationSeconds: number;
    estimatedBillableMinutes: number;
    estimatedCostCny: number;
    pricingRuleVersion: string;
    costEstimateAvailable: boolean;
    resultUrl: string;
    resultMimeType: string;
    resultBytes: number;
    errorCode: string;
    errorMessage: string;
    cloudProcessing: true;
    createdAt: string;
    startedAt: string;
    completedAt: string;
    updatedAt: string;
};

export type CreateVideoSubtitleEraseJobInput = { file: Blob; filename: string; projectId: string; canvasId: string; sourceNodeId: string; sourceAssetId?: string };

const token = () => useUserStore.getState().token;

export function getVideoSubtitleEraseCapabilities() {
    return apiGet<VideoSubtitleEraseCapabilities>("/api/v1/video-subtitle-erase/capabilities", undefined, token());
}

export function createVideoSubtitleEraseJob(input: CreateVideoSubtitleEraseJobInput) {
    const form = new FormData();
    form.append("file", input.file, input.filename);
    form.append("projectId", input.projectId);
    form.append("canvasId", input.canvasId);
    form.append("sourceNodeId", input.sourceNodeId);
    form.append("sourceAssetId", input.sourceAssetId || "");
    return apiPostForm<VideoSubtitleEraseJob>("/api/v1/video-subtitle-erase/jobs", form, token());
}

export function getVideoSubtitleEraseJob(jobId: string) {
    return apiGet<VideoSubtitleEraseJob>(`/api/v1/video-subtitle-erase/jobs/${encodeURIComponent(jobId)}`, undefined, token());
}

export function retryVideoSubtitleEraseJob(jobId: string) {
    return apiPostEmpty<VideoSubtitleEraseJob>(`/api/v1/video-subtitle-erase/jobs/${encodeURIComponent(jobId)}/retry`, token());
}
